### ISSUE TITLE
Changes to the stocktake detail view - carry over from 2278

### ISSUE THREAD

**Issue by roxy-dao** [openmsupply/open-msupply#2340]
Carry over from [comment](https://github.com/openmsupply/open-msupply/issues/2278#issuecomment-1747634708)

My suggestion summary of changes:

- Remove the grouping by item ( and all the logic that goes with that in the Detail table )
- Retain the per-item modal which shows all batches for the current item
- Add the highlight animation, or some other visual cue to show which batch you've clicked on when editing
- Change comment to be the speech-bubble-icon, show on hover column (possibly?)
and later make these changes:

filters as noted by MG
- by item
- expiry date

**Comment by DhanyaHerath** [comment]
Bumping that filtering by expiry in particular would be really helpful - we would then be able to find just the expired goods in the store, and dispose of them. Possibly something that could be in inventory adjustments instead though - @richardmoizeau @mark-prins @roxy-dao 

Maybe that part could be in a separate issue

**Comment by richardmoizeau** [comment]
Yes, definitely something that we want to include in Inventory Adjustments, possibly Returns too. Ideally a common prompt for these transactions would be ideal.
